static/src/javascripts-legacy/projects/facia/modules/ui/lazy-load-containers.js

```diff
@@ -16,30 +14,55 @@ define([
     var distanceBeforeLoad = detect.getViewport().height;
 
     return function () {
-        var $frontBottom = bonzo(qwery('.js-front-bottom')),
-            containers = qwery('.js-container--lazy-load'),
+        var containers = qwery('.js-container--lazy-load'),
             lazyLoad = throttle(function () {
                 if (containers.length === 0) {
                     mediator.off('window:throttledScroll', lazyLoad);
                 } else {
                     fastdom.read(function () {
-                        var scrollTop = window.pageYOffset,
-                            scrollBottom = scrollTop + bonzo.viewport().height,
-                            bottomOffset = $frontBottom.offset().top,
-                            $container;
-
-                        if (scrollBottom > bottomOffset - distanceBeforeLoad) {
-                            $container = bonzo(containers.shift());
-
-                            fastdom.write(function () {
-                                $container.removeClass('fc-container--lazy-load');
-                            });
-                        }
+                        var cs = split(containers, withinRange);
+
+                        containers = cs[1];
+
+                        fastdom.write(function () {
+                            cs[0].forEach(displayContainer);
+                        });
                     });
                 }
             }, 500);
 
         mediator.on('window:throttledScroll', lazyLoad);
         lazyLoad();
     };
+
+    // split :: [Element] -> (Element -> Bool) -> ([Element], [Element])
+    // invariant: { inA all fn ^ outA all (complement fn) | (inA, outA) = split xs fn }
+    // Split an array of elements into two arrays, one where element comply, one where they don't
+    function split(arr, testFn) {
```

Comment:
this feels to me like an in situ `reduce` at line 23 would be a more natural way to express this in js? something like:

```js
var containersInRange = containers.reduce(function(result, container) {
    result[inRange(container) ? 'in' : 'out'].push(container);
    return result;
}, { in: [], out: [] });
```
feels a lot more native and quickly understandable?

maybe it's a taste thing, and I don't want to meander off into the bike-shed so ignore if you think I am, but i'm not sure that abstracting it like this – and naming it after a scala method – really _is_ developer friendly. the context here _is_ js...